### ISSUE TITLE
Absolute local paths on windows not recognized

### ISSUE THREAD

**Issue by Dobatymo** [lycheeverse/lychee#972]
Sorry if I am using it wrong or it's duplicate report.

Running lychee on a local absolute path on Windows gives the following error:
```
>lychee.exe "C:\asd"
Error: Network error

Caused by:
    0: builder error for url (c:\asd): URL scheme is not allowed
    1: URL scheme is not allowed
```
Running it from that directory with `lychee.exe .` works!

**Comment by lebensterben** [comment]
this indeed is a bug.

Windows path is technically URL and the lychee tries to parse the argument as URL before parse it as a file path. And in the case the drive is mistaken as a scheme.

A simple workaround is to change the said order. But it won't work because someone may name a drive as "http". so "http:\\foo.bar\xyz" becomes both a legit Windows file path and a legit URL.

The true solution is:

Totally ban positional arguments. If you want to supply a file path, use --file, if you want to supply a glob, use --glob, etc.

**Comment by Dobatymo** [comment]
Windows drive names can only be letters `a-z`, so `http:\` would not be a valid Windows path. URLs would have to start with a double slash, which Windows paths don't (ignoring network drives like `\\server\path\` for now...

But yeah, I agree that paths/URLs are a mess and it's best to explicitly distinguish them!

**Comment by lebensterben** [comment]
in my made-up example `http:\foo.bar\xyz`:

- http is the drive name
- \foo.bar\xyz is an absolute path

> URLs would have to start with a schema and double slash

False. It doesn't have to contain double slash. Examples 

- `mailto:foo@bar.com`
- `foo:bar`

**Comment by Dobatymo** [comment]
Yes sorry, I was incorrect about the double slashes. However Windows doesn't really have drive names, only drive letters afaik.

**Comment by lebensterben** [comment]
okay after some research it turns out you're right.

on windows the drive letter must be one of A-Z.

(for example see https://learn.microsoft.com/en-us/windows-server/storage/disk-management/change-a-drive-letter)

So we can indeed guess whether a positional argument is a absolute path with drive on Windows.

**Comment by mre** [comment]
Can we collect a few test cases and the expected outcome here?

**Comment by lebensterben** [comment]
we really should not abuse positional argument even though we can guess whether it's a file path or a URL.

I think in most cases people invoke lychee against specific files or a glob pattern, so the practical choice is to allow path and glob as the positional argument.

directly providing a URL to lychee is a niche case actually. it should either be done with a keyword argument or we simply remove this feature altogether.

**Comment by Silvenga** [comment]
FWIW, you can workaround this with using drive relative paths - which basically look like standard POSIX paths.

So, assuming the working directory is on the `C:` drive, you can use `/Users/...`. Windows is a lot more forgiving with paths than POSIX systems.

**Comment by mre** [comment]
## Summary
lychee fails to recognize absolute local paths on Windows, mistaking the drive letter for a URL scheme.

## Example
```
lychee.exe "C:\Users\username\document.md"
```
Expected: lychee checks the file
Actual: Error: Network error (URL scheme is not allowed)

## Proposed Fix

1. When parsing input, first check if the path starts with a single uppercase letter followed by a colon (e.g., "C:").
2. If it does, treat it as a Windows absolute path rather than attempting to parse it as a URL.

This approach will correctly handle Windows paths while still supporting URLs as inputs (which I use frequently). It might not be generic, but we haven't discovered a lot of edge-cases like this, so I think we should be pragmatic here.

If you'd like to implement this fix, please submit a pull request with these changes.

**Comment by mre** [comment]
Tackling this in #1837. Feedback welcome! Also, if someone could try it, I'd be super thankful.

```sh
cargo install --git https://github.com/lycheeverse/lychee/ --branch absolute-local-windows-paths
```

**Comment by ofek** [comment]
Fixed?

**Comment by mre** [comment]
Not yet merged, but you can try the solution. It's as easy as running the command I posted.
